File: app/auth/roles.py
"""
Shared role constants — mirrors docs/ROLES.md.

This module is intended to be copied verbatim into ticket / opscenter / cloudmgmt
systems (or published as an internal pip package) so every service judges roles
against the same codes.
"""

ADMIN = "admin"

# Sales org
SALES = "sales"
SALES_MANAGER = "sales_manager"

# Cross-functional
FINANCE = "finance"
SUPPORT = "support"
AUDITOR = "auditor"
READONLY = "readonly"

ALL_ROLES = {
    ADMIN, SALES, SALES_MANAGER,
    FINANCE, SUPPORT, AUDITOR, READONLY,
}

# Convenience groups
WRITE_ROLES = {ADMIN, SALES, SALES_MANAGER}
READ_ROLES = ALL_ROLES  # 所有角色都至少能读
MANAGER_ROLES = {ADMIN, SALES_MANAGER}

File: docs/sso-templates/typescript/roles.ts
// Shared role constants — mirrors docs/ROLES.md.
// Keep this identical in every TS system (gongdan / opscenter / ...).

export const Roles = {
  ADMIN: 'admin',
  SALES: 'sales',
  SALES_MANAGER: 'sales_manager',
  OPS: 'ops',
  OPS_MANAGER: 'ops_manager',
  FINANCE: 'finance',
  SUPPORT: 'support',
  AUDITOR: 'auditor',
  READONLY: 'readonly',
} as const;

export type Role = (typeof Roles)[keyof typeof Roles];

export const WRITE_ROLES: Role[] = [Roles.ADMIN, Roles.SALES, Roles.SALES_MANAGER, Roles.OPS, Roles.OPS_MANAGER];
export const MANAGER_ROLES: Role[] = [Roles.ADMIN, Roles.SALES_MANAGER, Roles.OPS_MANAGER];
